docs/earth/forecast.md

```diff
@@ -8,3 +8,146 @@ kernelspec:
+For reference, see [this classic image](https://www.unidata.ucar.edu/presentations/caron/FmrcPoster.pdf) where the
+y-axis is the 'valid time', and the x-axis is the 'forecast reference or initialization time':
+![FMRC indexing schematic](./fmrc.png)
+
+- "Model Run" : a single model run.
+- "Constant Offset" : all values for a given lead time.
+- "Constant Forecast" : all forecasts for a given time in the future.
+- "Best Estimate" : A best guess stitching together the analysis or initialization fields for past forecasts with the latest forecast.
+
+## Highlights
+
+All of these patterns are "vectorized indexing", though generating the needed indexers is complicated by missing data.
+
+`ForecastIndex` encapsulates this logic and,
+
+1. Demonstrates how fairly complex indexing patterns can be abstracted away with a custom Index, and
+1. Again illustrates the value of a custom Index in persisting "state" or extra metadata (here the type of model used).
+
+## Example
+
+Read an example dataset -- downward short-wave radiation flux from the NOAA's HRRR forecast system ([High Resolution Rapid Refresh](https://rapidrefresh.noaa.gov/hrrr/)).
+
+```{code-cell}
+import rolodex.forecast
+import xarray as xr
+
+xr.set_options(display_expand_indexes=True, display_expand_data=False)
+
+cube = xr.tutorial.open_dataset("hrrr-cube")
+cube
+```
+
+### Assigning
+
+To assign we add a new variable: `valid_time = time + step`, "initialization time" + "time steps in to the future".
+
+```{code-cell}
+from rolodex.forecast import Model, ForecastIndex
+
+cube.coords["valid_time"] = rolodex.forecast.create_lazy_valid_time_variable(
+    reference_time=cube.time, period=cube.step
+)
+```
+
+With the `valid_time` array, we can illustrate the complex structure of this datacube.
+
+- green box illustrates the bounding box of valid forecast data, any white cells within this box are NaN --- no forecast was generated for that forecast time with that initialization time.
+- the horizontal black line "2025-01-02 21:00" has 4 valid forecasts, though the underlying data has 23 data points (mostly NaN).
+
+```{code-cell}
+---
+tags: [hide-input]
+---
+import numpy as np
+import matplotlib.pyplot as plt
+time = "2025-01-02T21:00"
+
+(
+    cube.dswrf
+    .isel(x=10, y=20)
+    .plot(
+        x="time",
+        y="valid_time",
+        cmap="plasma",
+        robust=True,
+        # edgecolors="#fefefe",
+        # linewidths=0.003,
+        aspect=2,
+        size=5,
+    )
+)
+plt.axhline(np.datetime64(time), color="k")
+plt.plot(
+    cube.time.data[[0, -1, -1, 0, 0]],
+    cube.valid_time.data[
+        [0, -1, -1, 0, 0],
+        [0, 0, -1, -1, 0],
+    ],
+    color="#32CD32",
+    lw=2,
+)
+```
+
+We will index out all forecasts for 2025-01-02 21:00, notice that there are 4 valid forecasts.
+
+And then assign `rolodex.ForecastIndex` to `time`, `step`, `valid_time`. We pass a custom kwarg `model` to indicate the type of forecast model used, here "HRRR".
```

Comment:
Looking briefly at this, it looks specifying `model=Model.HRRR` is required right? I thought at first it was maybe just an alternative more persistent label compared to what's already in `attrs`, but IIUC this allows the index code to implement logic for different forecast structures?